### ISSUE TITLE
[dynamo] aten.nonzero.default data-dependent output shape (59 models, 573 breaks)

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#117]
## Summary

Operator `aten.nonzero.default` produces a graph break because its output shape depends on input data. Affects compile pipelines that rely on `nonzero` (or implicitly via `tensor.where()`, `tensor.argmax/argmin` in some paths, or boolean masking patterns).

Currently bundled under umbrella issue #18 — splitting out per the umbrella-split policy (one root cause per issue → trackable progress).

## Break Reason

```
Dynamic shape operator
  Explanation: Operator `aten.nonzero.default`'s output shape depends on input Tensor data.
  Hint: Enable tracing of dynamic shape operators with `torch._dynamo.config.capture_dynamic_output_shape_ops = True`
  Developer debug context: aten.nonzero.default
```

**Total: 573 breaks across 59 distinct models, 116 (model,mode) pairs**

| Model | eval | train |
|---|---|---|
| AriaForConditionalGeneration | 5 | 5 |
| AriaModel | 5 | 5 |
| AudioFlamingo3ForConditionalGeneration | 4 | 1 |
| AyaVisionForConditionalGeneration | 2 | 2 |
| Cohere2VisionForConditionalGeneration | 5 | 5 |
| ConsistencyDecoderVAE | 1 | 2 |
| DbrxForCausalLM | 10 | 10 |
| DbrxModel | 8 | 8 |
| DeepseekVLForConditionalGeneration | 5 | 5 |
| Ernie4_5_VLMoeForConditionalGeneration | 5 | 5 |
| Ernie4_5_VL_MoeForConditionalGeneration | 5 | 5 |
| Florence2ForConditionalGeneration | 5 | 5 |
| Florence2Model | 3 | 3 |
| Gemma3ForConditionalGeneration | 5 | 5 |
| Gemma4-vision | 10 | - |
| Gemma4ForConditionalGeneration | 10 | 10 |
| Glm46VForConditionalGeneration | 5 | 5 |
| Glm4vForConditionalGeneration | 5 | 5 |
| Glm4vMoeForConditionalGeneration | 5 | 5 |
| GlmOcrForConditionalGeneration | 5 | 5 |
| GotOcr2ForConditionalGeneration | 5 | 5 |
| GotOcr2Model | 3 | 3 |
| InternVLForConditionalGeneration | 5 | 5 |
| InternVLModel | 3 | 3 |
| JanusForConditionalGeneration | 5 | 5 |
| Lfm2VlForConditionalGeneration | 2 | 2 |
| LightOnOcrForConditionalGeneration | 2 | 2 |
| Llama4ForConditionalGeneration | 4 | 4 |
| LlavaForConditionalGeneration | 2 | 2 |
| LlavaNextForConditionalGeneration | 2 | 2 |
| LlavaNextVideoForConditionalGeneration | 2 | 2 |
| LlavaOnevisionForConditionalGeneration | 2 | 2 |
| LongcatFlashForCausalLM | 10 | 10 |
| LongcatFlashModel | 8 | 8 |
| MiniCPM-V-ViT | 4 | 4 |
| Mistral3ForConditionalGeneration | 2 | 2 |
| OpenAIPrivacyFilterModel | 8 | 8 |
| OpenVoice-SynthesizerTrn-infer | 8 | - |
| Ovis2ForConditionalGeneration | 5 | 5 |
| PaddleOCRVLForConditionalGeneration | 5 | 5 |
| PaliGemmaForConditionalGeneration | 5 | 5 |
| PaliGemmaModel | 3 | 3 |
| Phi4MultimodalAudioModel | 4 | 2 |
| Qwen2AudioForConditionalGeneration | 2 | 1 |
| Qwen2VLForConditionalGeneration | 5 | 5 |
| Qwen2_5OmniThinkerForConditionalGeneration | 5 | 5 |
| Qwen2_5_VLForConditionalGeneration | 6 | 6 |
| Qwen3OmniMoeThinkerForConditionalGeneration | 7 | 7 |
| Qwen3VLForConditionalGeneration | 8 | 8 |
| Qwen3VLMoeForConditionalGeneration | 8 | 8 |
| Qwen3_5ForConditionalGeneration | 5 | 5 |
| Qwen3_5MoeForConditionalGeneration | 5 | 5 |
| SwitchTransformersEncoderModel | 6 | 6 |
| SwitchTransformersForConditionalGeneration | 8 | 8 |
| SwitchTransformersModel | 8 | 8 |
| VideoLlama3ForConditionalGeneration | 2 | 2 |
| ViltModel | 4 | 4 |
| VipLlavaForConditionalGeneration | 2 | 2 |
| VitsModel | 7 | 13 |

Top source locations:
```
  337× unknown
  8× transformers/models/dbrx/modeling_dbrx.py:310
  8× transformers/models/ernie4_5_vl_moe/modeling_ernie4_5_vl_moe.py:1228
  8× transformers/models/longcat_flash/modeling_longcat_flash.py:206
  6× transformers/models/switch_transformers/modeling_switch_transformers.py:169
  6× transformers/models/switch_transformers/modeling_switch_transformers.py:171
  4× transformers/models/ernie4_5_vl_moe/modeling_ernie4_5_vl_moe.py:1340
  4× transformers/models/florence2/modeling_florence2.py:721
  4× transformers/models/glm46v/modeling_glm46v.py:220
  4× transformers/models/glm4v/modeling_glm4v.py:1063
```
## Resolution paths

1. **Per-model upstream patches** — refactor `nonzero` callsites to avoid data-dependent shapes (e.g. use `torch.where` with broadcasting, masking with multiplications, or explicit unbacked sym handling).
2. **Document `capture_dynamic_output_shape_ops=True` as recommended PT2 config** — many of these compile cleanly with the flag set; trade-off is more guards / less optimization.
3. **Default flip discussion** — should `capture_dynamic_output_shape_ops` default to `True`?

## Closure criterion

When `tools/sweep_compare.py` shows ZERO breaks matching `'aten.nonzero.default' in text` in the current sweep (excluding skip-listed models), close this issue.

Split out from #18.
